AppController/djinn.rb

```diff
@@ -5874,27 +5874,20 @@ def get_all_stats(secret)
           Djinn.log_debug("Getting HAProxy stats for app: #{app_name}")
           if app_name != "none"
             total_reqs, reqs_enqueued, collection_time = get_haproxy_stats(app_name)
-            # Create the apps hash with useful information containing HAProxy stats.
-            tries = MAX_RETRIES
-            begin
-              all_stats["apps"][app_name] = {
+            if collection_time != :no_backend
+              # Create the apps hash with useful information containing HAProxy stats.
+              begin
+                all_stats["apps"][app_name] = {
                   "language" => @app_info_map[app_name]["language"].tr('^A-Za-z', ''),
                   "appservers" => @app_info_map[app_name]["appengine"].length,
                   "http" => @app_info_map[app_name]["nginx"],
                   "https" => @app_info_map[app_name]["nginx_https"],
                   "total_reqs" => total_reqs,
                   "reqs_enqueued" => reqs_enqueued
-              }
-            rescue => exception
-              backtrace = exception.backtrace.join("\n")
-              message = "Unforseen exception: #{exception} \nBacktrace: #{backtrace}"
-              
-              if tries > 0
-                Djinn.log_warn("Try: #{tries} Unforseen exception: #{exception} \nBacktrace: #{backtrace} \n")
-                Kernel.sleep(SMALL_WAIT)
-                tries -= 1
-                retry
-              else
+                }
+              rescue => exception
+                backtrace = exception.backtrace.join("\n")
+                message = "Unforseen exception: #{exception} \nBacktrace: #{backtrace}"
                 Djinn.log_warn("Unable to get application stats and exceeded number of retries: #{message}")
```

Comment:
Remove the line 'and exceeded number of retries' as the retry logic is not here anymore.